### PR TITLE
Migration to clear all role arrays from organizations table json / ma…

### PR DESCRIPTION
**Pull request by katherinebernstein**
…pstore

### CODE REVIEW

File: src/main/kotlin/com/openlattice/mechanic/upgrades/ClearJSONOrganizationRoles.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    override fun upgrade(): Boolean {
+
+        val organizations = HazelcastMap.ORGANIZATIONS.getMap(toolbox.hazelcast)
+
+        logger.info("About to clear roles for organizations")
+
+        organizations.executeOnEntries(OrganizationEntryProcessor {
+            it.roles.clear()
```
Comment (makosblade) [inline src/main/kotlin/com/openlattice/mechanic/upgrades/ClearJSONOrganizationRoles.kt:25]: does this need a `.setValue()` ? 🤔

**Review by makosblade — APPROVED** [review]

File: src/main/kotlin/com/openlattice/mechanic/upgrades/ClearJSONOrganizationRoles.kt (hunk 1 above)
Comment (makosblade) [reply src/main/kotlin/com/openlattice/mechanic/upgrades/ClearJSONOrganizationRoles.kt:25]: Oh, nvm, I see it

**Review by makosblade — APPROVED** [review]

File: src/main/kotlin/com/openlattice/mechanic/upgrades/ClearJSONOrganizationRoles.kt (hunk 1 above)
Comment (you-leee) [inline src/main/kotlin/com/openlattice/mechanic/upgrades/ClearJSONOrganizationRoles.kt:25]: Is the plan to also delete roles field from Organization class?

**Review by you-leee — APPROVED** [review]

File: src/main/kotlin/com/openlattice/mechanic/upgrades/ClearJSONOrganizationRoles.kt (hunk 1 above)
Comment (katherinebernstein) [reply src/main/kotlin/com/openlattice/mechanic/upgrades/ClearJSONOrganizationRoles.kt:25]: Not right now -- but when reading out an organization it's always populated by loading the roles through SecurePrincipalsManager and then calling `organization.roles.addAll( ... )` which leads to problems when that original set of roles is never updated or maintained (this is what I was referencing yesterday in hardcore btw)

Comment (you-leee) [reply src/main/kotlin/com/openlattice/mechanic/upgrades/ClearJSONOrganizationRoles.kt:25]: Thanks for clarifying that!

Comment (katherinebernstein) [reply src/main/kotlin/com/openlattice/mechanic/upgrades/ClearJSONOrganizationRoles.kt:25]: np!

**Review by pckeyes — APPROVED** [review]

**Pull request merged**